agent/src/main.rs

```diff
@@ -27,17 +38,60 @@ async fn run_agent() -> Result<()> {
         .await
         .context(error::ClientCreate)?;
 
-    let mut agent = BrupopAgent::new(k8s_client);
+    // Get node and bottlerocketnode names
+    let associated_node_name = env::var("MY_NODE_NAME").context(error::GetNodeName)?;
+    let associated_bottlerocketnode_name = format!("brn-{}", associated_node_name);
 
-    // Create a bottlerocketnode (custom resource) if associated bottlerocketnode does not exist
-    if !agent.check_node_custom_resource_exists().await? {
-        agent.create_metadata_custom_resource().await?;
-    }
+    // Generate reflector to watch and cache BottlerocketNodes
+    let brns = Api::<BottlerocketNode>::namespaced(k8s_client.clone(), NAMESPACE);
+    let brn_lp = ListParams::default()
+        .fields(format!("metadata.name={}", associated_bottlerocketnode_name).as_str());
+    let brn_store = reflector::store::Writer::<BottlerocketNode>::default();
+    let brn_reader = brn_store.as_reader();
+    let brn_reflector = reflector::reflector(brn_store, watcher(brns, brn_lp));
+    let brn_drainer = try_flatten_touched(brn_reflector)
+        .filter_map(|x| async move { std::result::Result::ok(x) })
+        .for_each(|_brn| {
+            // TODO: utilize tracing event
+            log::info!("Processed event for BottlerocketNodes");
```

Comment:
I would reduce the log level to `debug!` here, as I believe these events are quite noisy.